Release: FourKitchens Sprint 10

callinmullaney opened the pull request:
* Mega menu V1 keyboard improvements - https://fourkitchens.atlassian.net/browse/SHS-22

* Update class for breakout box color - https://fourkitchens.atlassian.net/browse/SHS-55

# READY FOR REVIEW

## Summary
- Improves keyboard navigation and focus styles for the mega menu

## Need Review By (Date)
9/28

## Urgency
low

## Steps to Test
1. Run `lando blt drupal:sync --site=music` or setup the music site on your local as this site in particular uses the mega menu
2. Using the keyboard, tab through the main navigation and ensure that any menu item that is a link and a button (Training for me on Traditional) that the button now receives an outline
3. Edit any page and add `Text Area` component. Verify there is a "Breakout Box" option under the "Styles Dropdown"


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

ryanhagerty committed:
Shs 22 megamenu v1 wcag (#1183)

* shs-22: start with focus

* shs-22: fix outline accessibility issue for megamenu v1

su-sws-jira commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1184-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-colorful-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-traditional-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/633106bbd65209e3aa08aa3d

ryanhagerty committed:
shs-55: edit breakout box wysiwyg class and color (#1181)

* shs-55: edit breakout box wysiwyg class and color

* shs-55: update compiled css

* shs-55: update utility class mark down file

su-sws-jira commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1184-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-colorful-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-traditional-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/633106bbd65209e3aa08aa3d

callinmullaney commented:
@pookmish This PR is ready for you when you have a moment. Mostly small FE changes. However, we did need to update some of the text formatting config in order to accommodate for the new BreakBox feature for WYSIWYG.

pookmish commented on config/default/filter.format.basic_html.yml:
```diff
@@ -57,7 +57,7 @@ filters:
     weight: -50
     settings:
-      allowed_html: '<b> <i> <em> <strong> <cite> <blockquote cite> <code> <ul type> <ol start type> <li> <dl> <dt> <dd> <h2 id class> <h3 id class> <h4 id class> <h5 id class> <h6 id class> <img src alt data-entity-type data-entity-uuid data-align data-caption class height width> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-al ...
+      allowed_html: '<b> <i class=""> <em> <strong> <cite> <blockquote cite> <code> <ul type> <ol start type> <li> <dl> <dt> <dd> <h2 id class=""> <h3 id class=""> <h4 id class=""> <h5 id class=""> <h6 id class=""> <img src alt data-entity-type data-entity-uuid data-align data-caption height width class=""> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-emb ...
```
The allowed attributes needs to be fixed.
Related to this issue: https://www.drupal.org/project/drupal/issues/2710427

pookmish commented on config/default/filter.format.basic_html_without_media.yml:
```diff
@@ -17,7 +17,7 @@ filters:
     weight: -10
     settings:
-      allowed_html: '<em> <strong> <cite> <blockquote cite> <code> <ul type> <ol start type> <li> <dl> <dt> <dd> <h2 id class> <h3 id class> <h4 id class> <h5 id class> <h6 id class> <table class> <caption class> <tbody class> <thead> <tfoot> <th scope class> <td class> <tr> <a id href name target hreflang data-entity-substitution data-entity-type data-entity-uuid title class> <p class> <div clas ...
+      allowed_html: '<em> <strong> <cite> <blockquote cite> <code> <ul type> <ol start type> <li> <dl> <dt> <dd> <h2 id class=""> <h3 id class=""> <h4 id class=""> <h5 id class=""> <h6 id class=""> <caption class=""> <tbody class=""> <thead> <tfoot> <th scope class=""> <td class=""> <tr> <div class=""> <br> <i class=""> <hr> <table class=" hs-table--borderless"> <a id href name target hreflang da ...
```
The allowed attributes needs to be fixed.
Related to this issue: https://www.drupal.org/project/drupal/issues/2710427

pookmish commented on config/default/filter.format.full_html.yml:
```diff
@@ -55,7 +55,7 @@ filters:
     weight: -43
     settings:
-      allowed_html: '<em> <strong> <cite> <blockquote cite> <code> <ul type> <ol start type> <li> <dl> <dt> <dd> <h2 id> <h3 id> <h4 id> <h5 id> <h6 id> <table class="decanter-table--borderless"> <caption> <tbody> <thead> <tfoot> <th> <td> <tr> <a href hreflang data-entity-substitution data-entity-type data-entity-uuid title class="decanter-button decanter-button--big decanter-button--secondary d ...
+      allowed_html: '<em> <strong> <cite> <blockquote cite> <code> <ul type> <ol start type> <li> <dl> <dt> <dd> <h2 id> <h3 id> <h4 id> <h5 id> <h6 id> <caption> <tbody> <thead> <tfoot> <th> <td> <tr> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align data-caption data-embed-button> <h1> <pre> <table class="decanter-table--bor ...
```
The allowed attributes needs to be fixed.
Related to this issue: https://www.drupal.org/project/drupal/issues/2710427

pookmish reviewed (changes requested):
If a PR is ready, please add me to the "Assignee" because I will get the notifications automatically and It'll show on my todo list.

callinmullaney committed:
fix: allowed formats bug that forces specific classes from style widget

su-sws-jira commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1184-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-colorful-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-traditional-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/633106bbd65209e3aa08aa3d

callinmullaney committed:
fix: allowed formats bug that forces specific classes from style widget

su-sws-jira commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1184-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-colorful-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-traditional-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/633106bbd65209e3aa08aa3d

callinmullaney committed:
fix: restore class limitation for full_html

su-sws-jira commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1184-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-colorful-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-traditional-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/633106bbd65209e3aa08aa3d

callinmullaney committed:
fix: restore class limitation for full_html

callinmullaney commented:
@pookmish Great find! Wasn't aware this was an issue with 9.4 or if this was how text_formats are being exported going forward. That being said, I addressed your feedback. Note: The `config/default/filter.format.full_html.yml` config had specific class restrictions I kept in place. Let me know if these should actually be removed as well.

su-sws-jira commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1184-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-colorful-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com
- https://hs-traditional-3zklyzqxsnr0qwltv1fvjzvfxja3l2sh.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/633106bbd65209e3aa08aa3d

pookmish reviewed (approved):
Thanks. Looks good now
